src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -433,69 +434,64 @@ public static void MoveDirectory(string sourceFullPath, string destFullPath)
 
         public static void RemoveDirectory(string fullPath, bool recursive)
         {
-            var di = new DirectoryInfo(fullPath);
-            if (!di.Exists)
+            if (!TryRemoveDirectory(fullPath, topLevel: true, throwWhenNotEmpty: !recursive))
             {
-                throw Interop.GetExceptionForIoErrno(Interop.Error.ENOENT.Info(), fullPath, isDirectory: true);
+                RemoveDirectoryRecursive(fullPath);
```

Comment:
this is confusing:
* a method named `Try*` can throw an exception instead of returning `false`
* it seems that the `recursive` argument is ignored and we always try a recursive removal when non-recursive has failed. I need to read `RemoveDirectoryRecursive` to find out it's not the case.

Reply:
I agree. What do you think about renaming `TryRemoveDirectory` to `TryRemoveEmptyDirectory`, and `RemoveDirectoryRecursive` to `RemoveDirectoryCore`?

Reply:
I've renamed the methods and added some comments.